### ISSUE TITLE
[no-type-alias] False positive with constructor aliases

### ISSUE THREAD

**Issue by mdouglass** [typescript-eslint/typescript-eslint#1191]
**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": { "project": "./tsconfig.json", "sourceType": "module" },
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/no-type-alias": ["error", { "allowCallbacks": "always" }]
  }
}

```

```TS
type ConstructorFunction = new (port: string) => any
```

**Expected Result**
No warning because allowCallbacks is set to always

**Actual Result**
  1:26  error  Type unhandled are not allowed  @typescript-eslint/no-type-alias

**Additional Info**
If the new is removed then this code passes without errors, so I presume it is because constructors are not handled. It is possible that this should be a new option (allowConstructors) rather than rolled into allowCallbacks.

**Versions**

| package | version |
| ---------------------------------- | ------- |
| @typescript-eslint/eslint-plugin | 2.6.1 |
| @typescript-eslint/parser | 2.6.1 |
| TypeScript | 3.7.2 |
| ESLint | 6.6.0 |
| node | 13.1.0 |
| npm | 6.12.1 |

**Comment by a-tarasyuk** [comment]
@bradzacher Is it a bug? Or need to add the new option `allowConstructors`?

**Comment by bradzacher** [comment]
It's a bit of both.

It's a case that was never thought of, hence it's marked as "unhandled"
https://github.com/typescript-eslint/typescript-eslint/blob/96d1cc3039e2c6402d60ed5443cd392232c23e0b/packages/eslint-plugin/src/rules/no-type-alias.ts#L239-L240

But it also doesn't really fit into any of the existing options. The closest would probably be `allowCallbacks`, but it's still not quite right.

So it's a bug that needs to be handled with a new option.
